src/main.js

```diff
@@ -25,79 +25,72 @@ const sortingItems = generateSortingItems();
 const renderWaypoint = (tripElement, waypoint) => {
   const waypointComponent = new WaypointView(waypoint);
   const waypointFormComponent = new WaypointFormView(waypoint);
-
-  const replaceWaypointToWaypointForm = () => {
-    tripElement.replaceChild(waypointFormComponent.getElement(), waypointComponent.getElement());
-
-    const tripDetails = waypointFormComponent.getElement().querySelector(`.event__details`);
-
-    tripDetails.innerHTML = ``;
-
-    render(tripDetails, new WaypointOffersView(waypoint.offers).getElement(), RenderPosition.AFTERBEGIN);
-    render(tripDetails, new WaypointDestinationView(waypoint.destination).getElement(), RenderPosition.BEFOREEND);
-  };
-
-  const replaceWaypointFormToWaypoint = () => {
-    tripElement.replaceChild(waypointComponent.getElement(), waypointFormComponent.getElement());
-  };
-
-  const waypointComponentBtn = waypointComponent.getElement().querySelector(`.event__rollup-btn`);
-
   const onEscKeyDown = (event) => {
     if (event.key === `Esc` || event.key === `Escape`) {
       event.preventDefault();
-      replaceWaypointFormToWaypoint();
+      replace(waypointComponent, waypointFormComponent);
+      document.removeEventListener(`click`, onClickWaypointBtn);
+      document.removeEventListener(`keydown`, onEscKeyDown);
+    }
+  };
+  const onClickWaypointBtn = (event) => {
+    const currentWaypointRollupBtn = waypointComponent.getElement().querySelector(`.event__rollup-btn`);
```

Comment:
- Сокращение `Btn` нарушает [критерий Б4](https://up.htmlacademy.ru/ecmascript/13/criteries#b4).
- Использование `component.getElement().querySelector('...')` нарушает [критерий Д26](https://up.htmlacademy.ru/ecmascript/13/criteries#d26).

Reply:
Про Д26, например так:
```suggestion
  const onClickWaypointBtn = (event) => {
    const currentWaypointRollupButton = waypointComponent.rollupButton;
```
и в waypoint.js в конструкторе:
constructor(waypoint) {
    ...
    this.rollupButton = this.getElement().querySelector(`.event__rollup-btn`);
}
Или как-то по-другому?